tests/test_core/test_events.py

```diff
@@ -353,3 +353,232 @@ def test_rotate(
+            "eID": [1, 2],
+            "gameclock": [1.1, 2.2],
+        }
+    )
+    start_events = pd.DataFrame(
+        {
+            "eID": [1],
+            "gameclock": [1.1],
+        }
+    )
+    end_events = pd.DataFrame(
+        {
+            "eID": [2],
+            "gameclock": [2.2],
+        }
+    )
+    assert all(data_minimal_full.events == full_events)
+    assert all(data_minimal_full_arguments.events == full_events)
+
+    assert all(data_minimal_end_sliced.events == start_events)
+
+    assert all(data_minimal_start_sliced.events == end_events)
+
+
+# tests for slice with inplace=True
+@pytest.mark.unit
+def test_slice_inplace(
+    example_events_data_minimal,
+    example_events_data_frameclock,
+) -> None:
+
+    # Arrange
+    data_minimal = Events(example_events_data_minimal)
+    data_frameclock = Events(example_events_data_frameclock)
+
+    # Act
+    data_minimal.slice(start=10, inplace=True)
+    data_frameclock.slice(start=15, slice_by="frameclock", inplace=True)
+
+    # Assert
+    frameclock_end_events = pd.DataFrame(
+        {
+            "eID": [2],
+            "gameclock": [0.2],
+            "frameclock": [16.7],
+        }
+    )
+    assert len(data_minimal.events) == 0
+    assert all(data_frameclock.events == frameclock_end_events)
+
+
+# tests slice using the frameclock column
+@pytest.mark.unit
+def test_slice_frameclock(
+    example_events_data_frameclock,
+):
+    # Arrange
+    data_frameclock = Events(example_events_data_frameclock)
+
+    # Act
+    data_frameclock_sliced = data_frameclock.slice(
+        end=15, slice_by="frameclock", inplace=False
+    )
+
+    # Assert
+    frameclock_start_events = pd.DataFrame(
+        {
+            "eID": [1],
+            "gameclock": [0.1],
+            "frameclock": [12.4],
+        }
+    )
+    assert all(data_frameclock_sliced.events == frameclock_start_events)
+
+
+# tests slice for objects containing None
+@pytest.mark.unit
+def test_slice_none(
+    example_events_data_minimal_none,
+):
+    # Arrange
+    data_none = Events(example_events_data_minimal_none)
+
+    # Act
+    data_none_sliced = data_none.slice(start=0, end=10, inplace=False)
+    data_none.slice(start=1.1, end=1.2, inplace=True)
+
+    # Assert
+    data_none_events = pd.DataFrame(
+        {
+            "eID": [None],
+            "gameclock": [1.1],
+        }
+    )
+    assert all(data_none_sliced.events == data_none_events)
+    assert all(data_none.events == data_none_events)
+
+
+# tests get_event_stream with and without frameclock (latter should fail)
+@pytest.mark.unit
+def test_get_event_stream_frameclock(
+    example_events_data_minimal,
+    example_events_data_frameclock,
+):
+    # Arrange
+    data_minimal = Events(example_events_data_minimal)
+    data_frameclock = Events(example_events_data_frameclock)
+
+    # Act
+    try:
+        event_stream_minimal = data_minimal.get_event_stream()
+    except ValueError:
+        event_stream_minimal = None
```

Comment:
There is a standard way of testing exceptions:
https://docs.pytest.org/en/6.2.x/assert.html